### PR TITLE
Avoids division by zero in land soil hydrology

### LINKED ISSUE #2583: Variable soil depth: Divided by zero bug
When the variable soil depth is set so that interface depth is equal to water depth. Namely, `zwtmm(c)=zimm(c,j)` in the following line of code: https://github.com/E3SM-Project/E3SM/blob/master/components/clm/src/biogeophys/SoilWaterMovementMod.F90#L494, it became divided by zero. 

I will propose the following fix:
```
delta_z_zwt= zwtmm(c)-zimm(c,j)
if(delta_z_zwt==0._r8) delta_z_zwt= 1 !make a hypothetical thin layer (1 mm)
temp0 = (((sucsat(c,j)+delta_z_zwt)/sucsat(c,j)))**(1._r8-1._r8/bsw(c,j))
vol_eq(c,j+1) = -sucsat(c,j)*watsat(c,j)/(1._r8-1._r8/bsw(c,j))/(delta_z_zwt)*(tempi-temp0)
```

Let me know your thoughts @bishtgautam

### PR DESCRIPTION
**Pull request by xuchongang**
Fixes division by zero when water table depth exactly equal the soil interface depth
by assuming the soil is fully saturated.

Fixes #2583

[BFB]

### CODE REVIEW

File: components/clm/src/biogeophys/SoilWaterMovementMod.F90 [hunk 1]
```diff
@@ -490,8 +491,13 @@ subroutine soilwater_zengdecker2009(bounds, num_hydrologyc, filter_hydrologyc, &
          j = nlev2bed(c)
          if(jwt(c) == nlevbed) then 
             tempi = 1._r8
-            temp0 = (((sucsat(c,j)+zwtmm(c)-zimm(c,j))/sucsat(c,j)))**(1._r8-1._r8/bsw(c,j))
-            vol_eq(c,j+1) = -sucsat(c,j)*watsat(c,j)/(1._r8-1._r8/bsw(c,j))/(zwtmm(c)-zimm(c,j))*(tempi-temp0)
+	    delta_z_zwt= zwtmm(c)-zimm(c,j)
```
Comment (bishtgautam) [inline components/clm/src/biogeophys/SoilWaterMovementMod.F90]: Please indent the code

Comment (xuchongang) [reply components/clm/src/biogeophys/SoilWaterMovementMod.F90]: @bishtgautam ,  the indent is correct when I open the file in  my folder for open using the online editing tool . This is only view bug from github. Let me know if that is not the case  for you.

**Comment by bishtgautam** [comment]
@xuchongang, Thanks for the new commit, but other lines also need indenting.

**Comment by xuchongang** [comment]
@bishtgautam , I have updated the indent request and let me know if you have further concerns.

**Comment by xuchongang** [comment]
@bishtgautam , it is quite tricky for the indentation. It seems there is different types of spaces. I've used the online edits to change and seems now the indentation is fine.

**Comment by bishtgautam** [comment]
@xuchongang, Shouldn't this PR be non-BFB?

**Comment by rljacob** [comment]
This was merged to next and found to be non-BFB.    Should it be?

**Comment by rljacob** [comment]
I'm closing this.   Reopen when/if the branch is fixed.

**Pull request closed without merging**